numpy/core/src/umath/ufunc_object.c

```diff
@@ -126,94 +126,93 @@ PyUFunc_clearfperr()
 }
 
 /*
- * This function analyzes the input arguments
- * and determines an appropriate __array_prepare__ function to call
- * for the outputs.
- * Assumes subok is already true if check_subok is false.
- *
- * If an output argument is provided, then it is prepped
- * with its own __array_prepare__ not with the one determined by
- * the input arguments.
- *
- * if the provided output argument is already an ndarray,
- * the prepping function is None (which means no prepping will
- * be done --- not even PyArray_Return).
- *
- * A NULL is placed in output_prep for outputs that
- * should just have PyArray_Return called.
+ * This function analyzes the input arguments and determines an appropriate
+ * method (__array_prepare__ or __array_wrap__) function to call, taking it
+ * from the input with the highest priority.
```

Comment:
Commend should indicate that 'NULL' means "no method found", not "an error occured" (which cannot happen at all right now)